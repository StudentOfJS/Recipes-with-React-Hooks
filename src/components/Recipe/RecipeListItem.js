import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { onEnterKey } from '../../utils';

const RecipeListItem = memo(
  ({ id, name, category, favorited, onClickOrPress, onFavorite }) => {
    const onKeyPress = onEnterKey(onFavorite);
    return (
      <li
        key={id}
        className="py2 border-bottom border-bottom-dashed pointer flex"
      >
        <span
          className="checkbox-container"
          role="checkbox"
          aria-checked={favorited}
          aria-label="check box to favorite recipe"
          id={id}
          onClick={onFavorite}
          tabIndex={id + id - 1}
          onKeyPress={onKeyPress}
        >
          <input
            aria-label="check box to favorite recipe"
            type="checkbox"
            checked={favorited}
            onChange={onFavorite}
          />
          <span className="checkmark" />
        </span>
        <button
          type="button"
          value={id}
          onClick={onClickOrPress}
          onKeyPress={onClickOrPress}
          className="menu-button"
          tabIndex={id + id}
        >
          {`${name} - ${category}`}
        </button>
      </li>
    );
  },
);

RecipeListItem.propTypes = {
  id: PropTypes.number.isRequired,
  category: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  favorited: PropTypes.bool.isRequired,
  onClickOrPress: PropTypes.func.isRequired,
  onFavorite: PropTypes.func.isRequired,
};

RecipeListItem.defaultProps = {
  id: 0,
  category: '',
  name: '',
  favorited: false,
  onClickOrPress: () => {},
  onFavorite: () => {},
};

export default RecipeListItem;
