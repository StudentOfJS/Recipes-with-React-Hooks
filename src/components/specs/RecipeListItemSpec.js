import React from 'react';
import renderer from 'react-test-renderer';
import RecipeListItem from '../Recipe/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'test recipe',
  category: 'test category',
};

describe('<Header />', () => {
  test('should not break if no recipe props passed', () => {
    const component = renderer.create(<RecipeListItem />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should correctly render recipe', () => {
    const component = renderer.create(<RecipeListItem {...testRecipe} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render favorite state', () => {
    const component = renderer.create(
      <RecipeListItem favorited {...testRecipe} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
