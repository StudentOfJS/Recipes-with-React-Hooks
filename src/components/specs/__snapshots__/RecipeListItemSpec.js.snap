// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should correctly render recipe 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer flex"
>
  <span
    aria-checked={false}
    aria-label="check box to favorite recipe"
    className="checkbox-container"
    id={1}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="checkbox"
    tabIndex={1}
  >
    <input
      aria-label="check box to favorite recipe"
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </span>
  <button
    className="menu-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={2}
    type="button"
    value={1}
  >
    test recipe - test category
  </button>
</li>
`;

exports[`<Header /> should not break if no recipe props passed 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer flex"
>
  <span
    aria-checked={false}
    aria-label="check box to favorite recipe"
    className="checkbox-container"
    id={0}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="checkbox"
    tabIndex={-1}
  >
    <input
      aria-label="check box to favorite recipe"
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </span>
  <button
    className="menu-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
    type="button"
    value={0}
  >
     - 
  </button>
</li>
`;

exports[`<Header /> should render favorite state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer flex"
>
  <span
    aria-checked={true}
    aria-label="check box to favorite recipe"
    className="checkbox-container"
    id={1}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="checkbox"
    tabIndex={1}
  >
    <input
      aria-label="check box to favorite recipe"
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkmark"
    />
  </span>
  <button
    className="menu-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={2}
    type="button"
    value={1}
  >
    test recipe - test category
  </button>
</li>
`;
